oeps/oep-0025-proc-incremental-improvements.rst

```diff
@@ -0,0 +1,247 @@
+==================================
+OEP-0025: Incremental Improvements
+==================================
+
++-----------------+----------------------------------------------------------+
+| OEP             | :doc:`OEP-0025 <oep-0025-proc-incremental-improvements>` |
++-----------------+----------------------------------------------------------+
+| Title           | Incremental Improvements                                 |
++-----------------+----------------------------------------------------------+
+| Last Modified   | 2018-06-13                                               |
++-----------------+----------------------------------------------------------+
+| Authors         | Jeremy Bowman <jbowman@edx.org>                          |
++-----------------+----------------------------------------------------------+
+| Arbiter         | Adam Stankiewicz <astankiewicz@edx.org>                  |
++-----------------+----------------------------------------------------------+
+| Status          | Draft                                                    |
++-----------------+----------------------------------------------------------+
+| Type            | Process                                                  |
++-----------------+----------------------------------------------------------+
+| Created         | 2018-06-13                                               |
++-----------------+----------------------------------------------------------+
+| `Review Period` | 2018-06-14 - 2018-06-29                                  |
++-----------------+----------------------------------------------------------+
+| `Resolution`    |                                                          |
++-----------------+----------------------------------------------------------+
+
+Abstract
+========
+
+Proposes a process for achieving large and/or long-term Open edX project goals
+via the definition and distributed completion of many small, straightforward
+development tasks.
+
+Motivation
+==========
+
+There are often relatively straightforward changes to the Open edX code base
+which prove difficult to accomplish due to the sheer amount of code involved.
+These are not major new features or complex refactorings of the code, but
+rather things like:
+
+* Support a new version of Python or Django
+* Replace all uses of certain deprecated code with its preferred replacement
+* Find all instances of a particular antipattern and fix them
+
+Such projects can involve changing hundreds of files and thousands of lines of
+code...but they are fundamentally composed of dozens of routine, localized
+changes which can be made independently and without any particular prior
+knowledge of how the Open edX code is structured.  It is very difficult for
+any one organization to prioritize tackling one of these projects by itself,
+but once broken down into very small pieces, the benefits of distributing
+them across the entire community become rapidly apparent.
+
+Specification
```

Comment:
Is there a recommendation on who and how quickly the contributing PR would be reviewed?  Is it expected that the creator (individual and/or team) of the INCR ticket would be the reviewer?  Or would it go through the OSPR review triage process?

Reply:
Some of the constraints on allowable tickets are targeted at making them much easier to review.  My hope is that we can make review of INCR tickets something that can be injected in the middle of an active sprint without significantly impacting the sprint objectives.  We should really be trying to prioritize such reviews as something to be done by an appropriate approved reviewer within 1-2 business days of getting both the PR and the contributor's CLA.  (Don't drop what you're in the middle of doing to review it, but don't get deep into something the next morning without finishing any pending reviews first.)  I'll see if I can get enough consensus around that to put some kind of SLA guidance in the OEP.  I was thinking they would go through OSPR triage, maybe with preferential treatment due to being small and easily completed.

Having a designated review team for each ticket could be a good idea for facilitating triage and avoiding blocks on a designated individual reviewer who's on vacation or otherwise preoccupied.